### PR TITLE
chore(exporters/elasticsearch-exporter): use TestContainers for IT

### LINKED ISSUE #3056: Migrate ES exporter integration tests to use TestContainers
**Description**

We currently use a library to start an embedded Elasticsearch instance for our integration tests; unfortunately it does not support Elasticsearch >= 7.x. We anyway wanted to migrate those tests to use [TestContainers](https://testcontainers.org), which will allow us to use the official ES docker images as base, which can be configured on the fly as we need.

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

Migrates ES tests to use [TestContainers](https://testcontainers.org). This is just a plain migration, and any cleanup/improvements should be done in #3055.

## Related issues

Closes #3056

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Comment by npepinpe** [comment]
For some reason the container is not reachable when running the tests on the `maven:3.6.0-jdk-11` but it works fine on `maven:3.6.0-jdk-8`? I'll try to figure out why's that. Shouldn't change the code, it's probably just a CI thing =/

**Comment by npepinpe** [comment]
Alright I figured out the issue thanks to the TestContainers team, we weren't provisioning enough memory to our docker-in-docker container. This is something I'll take up with the SRE team to get their input on what would be best practices here.

**Comment by menski** [comment]
@Zelldon can you please review this PR next week.
@npepinpe can you please rebase this branch to solve the conflict

Thanks

File: .ci/podSpecs/distribution.yml [hunk 1]
```diff
@@ -66,8 +70,8 @@ spec:
...
       resources:
         limits:
-          cpu: 1
-          memory: 512Mi
+          cpu: 4
+          memory: 4Gi
         requests:
-          cpu: 1
-          memory: 512Mi
+          cpu: 4
```
Comment (Zelldon) [inline .ci/podSpecs/distribution.yml]: Is this really necessary?

File: exporters/elasticsearch-exporter/pom.xml [hunk 2]
```diff
@@ -127,8 +127,15 @@
...
-      <artifactId>embedded-elasticsearch</artifactId>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>testcontainers</artifactId>
+      <scope>test</scope>
+    </dependency>
+
+    <!-- used for SocketUtil to grab open port -->
+    <dependency>
+      <groupId>io.zeebe</groupId>
+      <artifactId>zeebe-transport</artifactId>
```
Comment (Zelldon) [inline exporters/elasticsearch-exporter/pom.xml]: Could we just move SocketUtil then?

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
Looks good just some small questions

File: .ci/podSpecs/distribution.yml (hunk 1 above)
Comment (npepinpe) [reply .ci/podSpecs/distribution.yml]: Ah, for requests yeah, you're right it's not necessary.

File: exporters/elasticsearch-exporter/pom.xml (hunk 2 above)
Comment (npepinpe) [reply exporters/elasticsearch-exporter/pom.xml]: Probably a good idea; `zeebe-util`?

Comment (Zelldon) [reply exporters/elasticsearch-exporter/pom.xml]: Jep this was my thinking

**Review by ChrisKujawa — APPROVED** [review]
:+1:

**Pull request merged**